Sources/Target/Common/ThreadBase.cpp

```diff
@@ -30,5 +30,17 @@ ErrorCode ThreadBase::modifyRegisters(
   action(state);
   return writeCPUState(state);
 }
+
+ErrorCode ThreadBase::beforeResume() {
+  if (!_process->isAlive())
+    return kErrorProcessNotFound;
+
+  BreakpointManager *bpm = _process->hardwareBreakpointManager();
+  if (bpm != nullptr) {
+    bpm->enable();
```

Comment:
You could, I suppose, cache which threads it's already enabled on, and only enable it on new threads. But that doesn't seem worth the complexity.